src/methods/zonal.jl

```diff
@@ -131,6 +148,13 @@ end
 function _alloc_zonal(f, x, geoms, n; kw...)
     # Find first non-missing entry and count number of missing entries
     n_missing::Int = 0
+    # TODO: wrap this in try-catch, so that we can throw an intelligent error.
+    # There are a couple of scenarios where this can fail:
+    # 1. The function cannot accept an empty iterator
+    # 2. If passed a RasterStack, the function may not be able to handle layers.  
+    #    In this case you'll get a method error, where one of the arguments is a NamedTuple
+    #    that is the same as eltype(rs).  We should check for that and throw an informative error,
```

Comment:
Would be very slick

Reply:
Yeah, will try to see if we can do it!  This is probably not very performance sensitive so try catch overhead should be pretty low here.